### ISSUE TITLE
ACP Rule URL - Question

### ISSUE THREAD

**Issue by banzigaga** [daxm/fmcapi#18]
Hello, 

Is there a way to add an URL to an ACPRule?

Regards, 

Gabor

**Comment by parkerbrother1** [comment]
@banzigaga 
Yes, but right now you have to build it manually.  Eventually I will add a function in the ACPRule Class which takes a list of url object names and does the work for you, just like the other supported objects.

    url1 = URL(fmc=fmc1, name='_url1', url='asdf.org')
    url1.post()
    foo = [{"type":url1.type,"id":url1.id,"name":url1.name}]
    acprule1 = ACPRule(fmc=fmc1, acp_name=acp1.name)
    acprule1.urls = {"objects":foo}

**Comment by banzigaga** [comment]
Ok, thanks  @parkerbrother1 !

**Comment by daxm** [comment]
@parkerbrother1 have you made any progress on this?
Personally I like the separate method as it keeps things simple and the script writer doesn't need to be a programmer and/or need to know the underlying structure of the "objects" format.

**Comment by karmicgeezer** [comment]
I wrote some native Python to achieve this, I’ve been in the US for work for the last 5 weeks so swampped. 

I’ got back yesterday, now to work out how to add a new method to fmcapi.

**Comment by daxm** [comment]
@karmicgeezer,
Or post your code here and I can implement it.

**Comment by banzigaga** [comment]
Hello, just asking how about this enhancement?

**Comment by daxm** [comment]
@NetDevAutomate, any status on your "fix" for this?

**Comment by daxm** [comment]
Check out the release 20200615.0.  I believe we've finally integrated the URL addition into the AccessRule class.
